overlay.nix

```diff
@@ -13,11 +13,27 @@ final: prev: {
 
     # IncludeOS
     includeos = self.stdenv.mkDerivation rec {
+      enableParallelBuilding = true;
       pname = "includeos";
 
       version = "dev";
 
-      src = prev.pkgsStatic.lib.cleanSource ./.;
+      src = prev.pkgsStatic.lib.fileset.toSource {
+          root = ./.;
+          # Only include files needed by IncludeOS (not examples, docs etc)
+          fileset = prev.pkgsStatic.lib.fileset.unions [
+            ./src
+            ./api
+            ./cmake
+            ./deps
+            ./userspace
+            ./lib
+            ./default.nix
```

Comment:
Nix files are typically only needed at eval time. You should be able to remove them from this list and in the future be able to refactor the Nix files without having to do any rebuilds (as long as the result evaluates to the same derivations).